### PR TITLE
Rest jms check

### PR DESCRIPTION
**Pull request by elvisciotti**

### CODE REVIEW

**Comment by elvisciotti** [comment]
- REST listener requires a JMS group if an entity is detected in the response. If not, a clear exception is thrown (this would have made easier spotting the edit note save crashing). Exception messages of course skip this check
- groups set for all the endpoints (When missing). I've tested they are the right ones. That will avoid  potential recursion and crashing (happened once with user 103 activation)
- PUT endpoints aligned to return the changed record ID (default behaviour)

File: src/AppBundle/EventListener/RestInputOuputFormatter.php [hunk 1]
```diff
@@ -111,6 +123,20 @@ private function arrayToResponse($data, Request $request)
...
+        if (is_array($object)) {
+            foreach($object as $subObject) {
+                if ($this->containsEntity($subObject)) {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        return is_object($object) && strpos(get_class($object), 'Entity') !== false;
```
Comment (slizzio) [inline src/AppBundle/EventListener/RestInputOuputFormatter.php]: Is there not an interface we can use here? If we're using strpos we may as well var_export the whole object and call strpos(), and remove the loop

File: src/AppBundle/EventListener/RestInputOuputFormatter.php [hunk 2]
```diff
@@ -76,21 +76,28 @@ public function requestContentToArray(Request $request)
...
+     * 
+     *
      *
+     * @param $data
+     * @param Request $request
+     * @param bool $groupsCheck
      * @return Response
      */
-    private function arrayToResponse($data, Request $request)
+    private function arrayToResponse(array $data, Request $request, $groupsCheck = false)
```
Comment (slizzio) [inline src/AppBundle/EventListener/RestInputOuputFormatter.php]: should we have the checks on by default?

Comment (elvisciotti) [inline src/AppBundle/EventListener/RestInputOuputFormatter.php]: changed

File: src/AppBundle/EventListener/RestInputOuputFormatter.php (hunk 1 above)
Comment (elvisciotti) [inline src/AppBundle/EventListener/RestInputOuputFormatter.php]: I thought about that but var_export could have to deal with a very big object (hope not a recursion) therefore crashing the API

**Pull request merged**